store/sqlstore/command_store.go

```diff
@@ -44,24 +44,22 @@ func (s SqlCommandStore) CreateIndexesIfNotExists() {
 	s.CreateIndexIfNotExists("idx_command_delete_at", "Commands", "DeleteAt")
 }
 
-func (s SqlCommandStore) Save(command *model.Command) store.StoreChannel {
-	return store.Do(func(result *store.StoreResult) {
-		if len(command.Id) > 0 {
-			result.Err = model.NewAppError("SqlCommandStore.Save", "store.sql_command.save.saving_overwrite.app_error", nil, "id="+command.Id, http.StatusBadRequest)
-			return
-		}
+func (s SqlCommandStore) Save(command *model.Command) (*model.Command, *model.AppError) {
+	if len(command.Id) > 0 {
+		return nil, model.NewAppError("SqlCommandStore.Save", "store.sql_command.save.saving_overwrite.app_error", nil, "id="+command.Id, http.StatusBadRequest)
+	}
 
-		command.PreSave()
-		if result.Err = command.IsValid(); result.Err != nil {
-			return
-		}
+	command.PreSave()
+	if err := command.IsValid(); err != nil {
+		return nil, err
+	}
 
-		if err := s.GetMaster().Insert(command); err != nil {
-			result.Err = model.NewAppError("SqlCommandStore.Save", "store.sql_command.save.saving.app_error", nil, "id="+command.Id+", "+err.Error(), http.StatusInternalServerError)
-		} else {
-			result.Data = command
-		}
-	})
+	err := s.GetMaster().Insert(command)
+	if err != nil {
```

Comment:
nit: You can do it in one line if you want

Reply:
It can't be done in one line, unless we change the return type as `error` but currently return type is `*model.AppError`

Reply:
Sorry, what i meant was, something like `if err := s.GetMaster().Insert(command); err != nil {`

Reply:
Yeah, can do that.